[niha81] iP

niha81 opened the pull request:
# Woogie
> **_Aspire to be young and rich, tall and handsome? Woogie is what you need!_** - ME

### What is Woogie?
Woogie is **THE** task manager chatbot that keeps track of your tasks. It is:
- 📝 Text-based
- 🎯 Easy to use
- ⚡ Fast & efficient

### How to Get Started?

- [ ] Download the latest version from [here](https://github.com/niha81/ip/releases/tag/A-Jar).
- [ ] Run the JAR file.
- [ ] Start managing your tasks! 

### Features
1. ✅ Task management
2. ⏳ Deadlines & Events tracking
3. 🔍 Find tasks by keywords
4. 💾 Persistent storage for tasks

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

niha81 committed:
level 0 done

niha81 committed:
level 1 done

niha81 committed:
add, list

niha81 committed:
mark done

niha81 committed:
todo, event, deadline

niha81 committed:
handle errors

niha81 committed:
delete

niha81 committed:
automated text ui testing

niha81 committed:
added exceptions, fixed syntax

niha81 committed:
Implement file save and load

niha81 committed:
Merge branch-Level-7

niha81 committed:
convert dates and times

niha81 committed:
Merge branch-Level-8

niha81 committed:
Extract Ui, TaskList and Parser classes

niha81 committed:
Organise classes into packages

niha81 committed:
Merge branch 'add-gradle-support'

niha81 committed:
Merge tag 'A-MoreOOP'

niha81 committed:
Merge tag 'A-Packages'

niha81 committed:
Add JUnit tests

niha81 committed:
Create JAR file

niha81 committed:
Add JavaDoc comments to main classes

niha81 committed:
Follow coding standard

niha81 committed:
Add find function

niha81 committed:
Resolve merge conflicts from A-CodingStandard

niha81 committed:
Resolve merge conflicts from Level-9

h-b-k-nishi commented on src/main/java/woogie/task/Task.java:
```diff
@@ -0,0 +1,103 @@
+     * @param line The line from the file representing a task.
+     * @return The corresponding Task object, or null if the format is invalid.
+     */
+    public static Task fromFileFormat(String line) {
```
Could you make the method name a verb, such as 'createTaskFromFileFormat'?

h-b-k-nishi commented on src/main/java/woogie/command/Parser.java:
```diff
@@ -0,0 +1,143 @@
+     * @param tasks TaskList containing the tasks.
+     * @param ui    User interface for displaying messages.
+     */
+    public static void processCommand(String input, TaskList tasks, Ui ui) {
```
How about using a more specific method name, such as 'readAndExecuteCommand'?

h-b-k-nishi reviewed (commented):
LGTM. Your code is clear and well-structured!

h-b-k-nishi commented on src/main/java/woogie/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+ * A Deadline task has a specific date and time by which it must be completed.
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
```
How about using nouns like 'dueDateTime'?

h-b-k-nishi commented on src/main/java/woogie/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+ * An Event task has a specific start date/time and an end date/time.
+ */
+public class Event extends Task {
+    private LocalDateTime from;
```
How about using nouns like 'startTime' and 'endTime'?

niha81 committed:
Add GUI for chatbot

niha81 committed:
Add asserstions

niha81 committed:
Improve code quality

- Extract repetitive logic into helper methods for better SLAP
  adherence
- Reduce nested conditionals to prevent arrowhead anti-patterns
- Replace hardcoded values with constants for flexibility
- Improve error handling to provide clearer feedback to users

This makes the codebase clearer, easier to maintain, and more scalable
for future extensions.

niha81 committed:
Merge pull request #1 from niha81/branch-A-Assertions

Add asserstions

niha81 committed:
Merge branch 'master' into branch-A-CodeQuality

niha81 committed:
Merge pull request #2 from niha81/branch-A-CodeQuality

Improve code quality

niha81 committed:
Implement sorting extension

- Sort ToDos alphabetically
- Sort Deadlines chronologically
- Sort Events chronologically by start time

niha81 committed:
Improve GUI

niha81 committed:
Implement final changes

niha81 committed:
Add assertions

niha81 committed:
Edit guide

niha81 committed:
Update README.md
